edgelet/edgelet-docker/src/runtime.rs

```diff
@@ -636,85 +631,67 @@ impl ModuleRuntime for DockerModuleRuntime {
                 })
             });
 
-        #[cfg(not(windows))]
-        {
-            #[cfg(target_os = "linux")]
-            let uptime = {
-                let mut info: libc::sysinfo = unsafe { mem::zeroed() };
-                let ret = unsafe { libc::sysinfo(&mut info) };
-                if ret == 0 {
-                    info.uptime.try_into().unwrap_or_default()
-                } else {
-                    0
-                }
-            };
-            #[cfg(not(target_os = "linux"))]
-            let uptime = 0;
-
-            let mut system_info = sysinfo::System::new();
-            system_info.refresh_all();
-            let current_time = SystemTime::now()
-                .duration_since(UNIX_EPOCH)
-                .unwrap_or_default()
-                .as_secs();
-            let start_time = process::id()
-                .try_into()
-                .map(|id| {
-                    system_info
-                        .get_process_list()
-                        .get(&id)
-                        .map(|p| p.start_time())
-                        .unwrap_or_default()
-                })
-                .unwrap_or_default();
-
-            let used_cpu = system_info
-                .get_processor_list()
-                .iter()
-                .find(|p| p.get_name() == "cpu")
-                .map_or_else(|| -1.0, |p| p.get_cpu_usage());
-
-            let total_memory = system_info.get_total_memory() * 1000;
-            let used_memory = system_info.get_used_memory() * 1000;
-
-            let disks = system_info
-                .get_disks()
-                .iter()
-                .map(|disk| {
-                    DiskInfo::new(
-                        disk.get_name().to_string_lossy().into_owned(),
-                        disk.get_available_space(),
-                        disk.get_total_space(),
-                        String::from_utf8_lossy(disk.get_file_system()).into_owned(),
-                        format!("{:?}", disk.get_type()),
-                    )
-                })
-                .collect();
-
-            let result = docker_stats.map(move |stats: String| {
-                SystemResources::new(
-                    uptime,
-                    current_time - start_time,
-                    used_cpu.into(),
-                    used_memory,
-                    total_memory,
-                    disks,
-                    stats,
-                )
-            });
+        #[cfg(target_os = "linux")]
+        let uptime: u64 = {
+            let mut info: libc::sysinfo = unsafe { mem::zeroed() };
+            let ret = unsafe { libc::sysinfo(&mut info) };
+            if ret == 0 {
+                info.uptime.try_into().unwrap_or_default()
+            } else {
+                0
+            }
+        };
+        #[cfg(windows)]
+        let uptime: u64 = unsafe { winapi::um::sysinfoapi::GetTickCount() }.into();
```

Comment:
This is in milliseconds whereas the Linux one is in seconds. Should they not be consistent?

Reply:
yes